### PR TITLE
39 admin requests request

### PR DESCRIPTION
**Pull request by AhmedAbuSharar**

### CODE REVIEW

File: server/test/index.test.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
-test("test", () => {
+test('test', () => {
   expect(1).toBe(1);
 });
```
Comment (Raghad-Mezied) [inline server/test/index.test.ts]: delete this test

File: server/app.ts [hunk 2]
```diff
@@ -12,11 +11,13 @@ app.use([
...
   cookieParser(),
   express.urlencoded({ extended: false }),
+  express.json(),
 ]);
 
 app.set('port', process.env.PORT || 8080);
 
-app.get('/data', (req: Request, res:Response) => res.send('Hello There!'));
+app.use(router);
+// app.get('/data', (req: Request, res:Response) => res.send('Hello There!'));
```
Comment (Raghad-Mezied) [inline server/app.ts]: Delete the commented line

File: server/controllers/admin/requestStatus.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable camelcase */
```
Comment (Raghad-Mezied) [inline server/controllers/admin/requestStatus.ts]: fix eslint

File: server/controllers/admin/requestStatus.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  try {
+    const { status } = req.body;
+    const { requestId } = req.params;
+    if (status === 'Approved' || status === 'Pending' || status === 'Rejected') {
+      await requestStatusQuery(status, +requestId);
+      res.json({ message: 'successful' });
+    } else {
+      throw new CustomError(400, 'invalid input');
+    }
+  } catch (err : any) {
```
Comment (Raghad-Mezied) [inline server/controllers/admin/requestStatus.ts]: Use Joi validation to validate your status and requestId

File: server/database/seeds/buildSeeds.ts [hunk 5]
```diff
@@ -29,7 +29,6 @@ const buildSeeds = async () => {
   await salesHistoryModel.bulkCreate(salesHistory);
   await adminModel.bulkCreate(admin);
 };
-
-buildSeeds();
+if (process.env.NODE_ENV === 'development') { buildSeeds(); }
```
Comment (Raghad-Mezied) [inline server/database/seeds/buildSeeds.ts:32]: Format

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/controllers/admin/requestStatus.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import requestStatusQuery from '../../queries';
+import { CustomError } from '../../utils';
+
+const requestStatus = async (req : Request, res : Response, next : NextFunction) => {
+  try {
+    const { status } = req.body;
+    const { requestId } = req.params;
+    if (status === 'Approved' || status === 'Pending' || status === 'Rejected') {
+      await requestStatusQuery(status, +requestId);
+      res.json({ message: 'successful' });
```
Comment (Raghad-Mezied) [inline server/controllers/admin/requestStatus.ts]: edit the `message` to be `msg`, so you all have the same key

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/controllers/admin/requestStatus.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    const { status } = req.body;
+    const { requestId } = req.params;
+    const isValid = await requestStatusSchema.validateAsync({ status, requestId });
+    if (isValid) {
+      await requestStatusQuery(status, +requestId);
+      res.json({ msg: 'successful' });
+    } else {
+      throw new CustomError(400, 'invalid input');
+    }
```
Comment (Raghad-Mezied) [inline server/controllers/admin/requestStatus.ts]: If the data is not validated, then the validation function will throw the error and your catch will catch it. 
so if you are still in line 12 then your data is validated, which means no need to isValid and it doesn't return a boolean.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by Ibtisam-Hemmo — APPROVED** [review]

**Review by Nader-SH — APPROVED** [review]

**Pull request merged**